### PR TITLE
fix(data-list): add spacing to compact data-list buttons

### PR DESCRIPTION
**Pull request by sg00dwin**
fixes https://github.com/patternfly/patternfly/issues/7105

- Update DataList compact examples to use small buttons
- Add top and bottom padding to small buttons

Backstop tests
https://drive.google.com/drive/folders/1aKh3N42bPbMjsfkTTbGOrQr8zUO_YHYn?usp=sharing

**Before**
Screenshot 2024-11-01 at 6 13 52 PM

**After**
Screenshot 2024-11-01 at 6 13 32 PM

**Before**
Screenshot 2024-11-01 at 6 51 25 PM

**After**
Screenshot 2024-11-01 at 6 50 01 PM

### CODE REVIEW

File: src/patternfly/components/DataList/data-list-item-action.hbs [hunk 1]
```diff
@@ -6,6 +6,7 @@
     {{> @partial-block}}
   {{else}}
     {{> menu-toggle
+      menu-toggle--IsSmall=data-list--IsCompact
```
Comment (mcoker) [inline src/patternfly/components/DataList/data-list-item-action.hbs:9]: 👍

**Review by mcoker — COMMENTED** [review]
LGTM!! Can you also update the actions in this example to use the small variation? https://patternfly-pr-7189.surge.sh/components/data-list/html/expandable-compact/

One other thing, I love the hbs `data-list--IsCompact` parameter - totally optional or could be a follow up, but it would be great if that could set `.pf-m-compact` as well. Otherwise it's a little confusing that a `.pf-m-compact` data-list also needs `data-list--IsCompact` to attempt to resize its buttons. As far as I can tell there are only 4 compact data lists in our docs that would need to be updated.

**Review by srambach — APPROVED** [review]
Looks great! 👍🏻

**Review by mcoker — APPROVED** [review]
🔥

**Pull request merged**